temporalio/worker/_workflow_instance.py

```diff
@@ -1415,14 +1439,47 @@ def workflow_upsert_search_attributes(
                         else [update.value]
                     )
 
+    async def workflow_sleep(
+        self, duration: float, *, summary: Optional[str] = None
+    ) -> None:
+        user_metadata = (
+            temporalio.api.sdk.v1.UserMetadata(
+                summary=self._payload_converter.to_payload(summary)
+            )
+            if summary
+            else None
+        )
+        self._timer_impl(
```

Comment:
There is nothing awaited here so I don't think this call is going to properly block. Internally I think asyncio invokes that callback on timer complete, so we need to block here waiting on that callback. I'd be ok using an `asyncio.Event` that is awaited here and set in the callback (or a wait condition, I haven't sat to think about it tbh).

Reply:
Duh, thanks